server/src/test/java/io/crate/lucene/CommonQueryBuilderTest.java

```diff
@@ -786,4 +786,10 @@ public void test_nested_not_operators() {
         Query query = convert("not (y is not null)");
         assertThat(query).hasToString("+(+*:* -FieldExistsQuery [field=y])");
     }
+
+    @Test
+    public void test_not_operator_on_query_equivalent_to_null() {
+        Query query = convert("(y % null != 1)");
+        assertThat(query).hasToString("+(+*:* -((y % NULL) = 1)) +FieldExistsQuery [field=y]");
```

Comment:
Could have been better if the Lucene query generated is `MatchNoDocsQuery` but identifying `y % null != 1` equivalent to `null` is the responsibility of the normalizer.